libcxx/test/configs/cmake-bridge.cfg.in

```diff
@@ -30,3 +30,7 @@ config.substitutions.append(('%{target-include-dir}', '@LIBCXX_TESTING_INSTALL_P
 config.substitutions.append(('%{lib-dir}', '@LIBCXX_TESTING_INSTALL_PREFIX@/@LIBCXX_INSTALL_LIBRARY_DIR@'))
 config.substitutions.append(('%{module-dir}', '@LIBCXX_TESTING_INSTALL_PREFIX@/@LIBCXX_INSTALL_MODULES_DIR@'))
 config.substitutions.append(('%{test-tools-dir}', '@LIBCXX_TEST_TOOLS_PATH@'))
+
+# Available if building libc++ on top of llvm-libc.
+config.substitutions.append(('%{libc-dir}', '@LIBCXX_LIBC_LIBRARY_DIR@'))
+config.substitutions.append(('%{libc-include-dir}', '@LIBCXX_LIBC_INCLUDE_DIR@'))
```

Comment:
We want to keep the `cmake-bridge` as lean as possible, and adding "optional" substitutions like this goes against that principle.

It seems like a smell that you need to pass `%{libc-dir}/crt1.o -lc -lm` explicitly when linking: why do you need that? You didn't pass `-nostdlib` so I would expect Clang to use the default stdlib on the system. It looks like the underlying problem is really that your compiler is not configured properly to use the just-built llvm-libc and you're working around that in the libc++ testing configuration.

Reply:
This is a GPU, executing stuff this way is the exceptions exception. I actually used to link `-lc` by default but that turned out to be a huge pain to deal with when working with the build systems so I removed it. I could *possibly* make an argument like `-stdlib` and `-startfiles` that adds that in this case, since it should know where to find them. However these positive versions of those flags don't really exist, so it'd need to be a new clang flag that's really specific.

Reply:
I made https://github.com/llvm/llvm-project/pull/112025. If this lands will adding `-gpustartfiles` to the commands be acceptable?

Reply:
Either way is fine by me. I care mostly that the general-purpose libc++ configuration doesn't get impacted by this complexity, and on the side I'm trying to vouch for naive users trying to just do a hello world. I think the `ADDITIONAL_PARAMETERs` approach above is also acceptable in the meantime just to unblock you.

I think `-gpustartfiles` might be the better approach, but I'm not strongly attached to it. If you want to use `ADDITIONAL_PARAMETERS` for now, that's fine by me too.